src/libraries/System.Private.CoreLib/src/System/Number.DecimalIeee754.cs

```diff
@@ -0,0 +1,255 @@
+        static abstract int Bias { get; }
+        static abstract int CountDigits(TSignificand number);
+        static abstract TSignificand Power10(int exponent);
+        static abstract int NumberBitsEncoding { get; }
+        static abstract TValue G0G1Mask { get; }
+        static abstract TValue MostSignificantBitOfSignificandMask { get; }
+        static abstract TValue SignMask { get; }
+        static abstract int NumberBitsCombinationField { get; }
+        static abstract int NumberBitsExponent { get; }
+        static abstract TValue PositiveInfinityBits { get; }
+        static abstract TValue NegativeInfinityBits { get; }
+        static abstract TValue Zero { get; }
+        static abstract string OverflowMessage { get; }
+    }
+
+    internal interface IDecimalIeee754UnpackInfo<TSelf, TSignificand, TValue>
+        where TSelf : unmanaged, IDecimalIeee754UnpackInfo<TSelf, TSignificand, TValue>
+        where TSignificand : IBinaryInteger<TSignificand>
+        where TValue : IBinaryInteger<TValue>
+    {
+        static abstract TValue SignMask { get; }
+        static abstract TValue G0G1Mask { get; }
+        static abstract TValue G0ToGwPlus1ExponentMask { get; } //G0 to G(w+1)
+        static abstract TValue G2ToGwPlus3ExponentMask { get; } //G2 to G(w+3)
+        static abstract TValue GwPlus2ToGwPlus4SignificandMask { get; } //G(w+2) to G(w+4)
+        static abstract TValue GwPlus4SignificandMask { get; } //G(w+4)
+        static abstract int NumberBitsSignificand { get; }
+        static abstract int NumberDigitsPrecision { get; }
+        static abstract int Bias { get; }
+        static abstract TValue MostSignificantBitOfSignificandMask { get; }
+        static abstract int ConvertToExponent(TValue value);
+        static abstract TSignificand ConvertToSignificand(TValue value);
+        static abstract TSignificand Power10(int exponent);
+    }
+
+    internal static partial class Number
+    {
+        internal static TValue CalDecimalIeee754<TDecimal, TSignificand, TValue>(TSignificand significand, int exponent)
+            where TDecimal : unmanaged, IDecimalIeee754ConstructorInfo<TDecimal, TSignificand, TValue>
+            where TSignificand : IBinaryInteger<TSignificand>
+            where TValue : IBinaryInteger<TValue>
+        {
+            if (significand == TSignificand.Zero)
+            {
+                return TValue.Zero;
+            }
+
+            TSignificand unsignedSignificand = significand > TSignificand.Zero ? significand : -significand;
+
+            if ((unsignedSignificand > TDecimal.MaxSignificand && exponent >= TDecimal.MaxDecimalExponent)
+                || (unsignedSignificand == TDecimal.MaxSignificand && exponent > TDecimal.MaxDecimalExponent))
+            {
+                return significand > TSignificand.Zero ? TDecimal.PositiveInfinityBits : TDecimal.NegativeInfinityBits;
+            }
+
+            TSignificand ten = TSignificand.CreateTruncating(10);
```

Comment:
This should be moved down to within the `if`, so it isn't created unless its actually going to be used.